gpt_index/vector_stores/pinecone.py

```diff
@@ -74,7 +80,28 @@ def __init__(
             import pinecone  # noqa: F401
         except ImportError:
             raise ImportError(import_err_msg)
-        self._pinecone_index = cast(pinecone.Index, pinecone_index)
+
+        self._index_name = index_name
+        self._environment = environment
+        if pinecone_index is not None:
+            self._pinecone_index = cast(pinecone.Index, pinecone_index)
```

Comment:
i wonder if it would still be nicer to support arbitrary arguments (e.g. pinecone index) instead of "only" supporting arguments that we can serialize. 

makes designing a vector store abstraction way easier, and we don't have to maintain knowledge of all arguments that would have to go into the underlying pinecone index

Reply:
Both are possible now.